smart-contract-usage-in-javascript-sdk.md

```diff
@@ -0,0 +1,326 @@
+    put(state{ available_count = remaining_count })
+
+
+  // function to get the remaining count, can be statically called as it doesn't change state
+  public function available_count() : int =
+    state.available_count
+
+
+  // function only to be called by the owner to withdraw all balance that was sent to the contract
+  public stateful function owner_withdraw() : int=
+  
+    // requiring contract caller to be the registered owner
+    require(Call.caller == state.owner, "Only owner can withdraw")
+    let withdrawed_balance = Contract.balance
+    
+    // creating spend of the balance to the registered owner
+    Chain.spend(state.owner, Contract.balance)
+    
+    // returning the balance that was withdrawed
+    withdrawed_balance
+```
+
+## SDK Contracts Workflow
+
+
+### 1. Initialize SDK client
+
+Inside the main function, to create the SDK client referencing the æternity network node and the needed public and private keys, add after the previously added lines:
+
+```
+// initialize aeternity sdk client
+const client = await Universal({
+    url: 'https://sdk-testnet.aepps.com', //replace with https://sdk-mainnet.aepps.com for mainnet
+    internalUrl: 'https://sdk-testnet.aepps.com', //replace with https://sdk-mainnet.aepps.com for mainnet
+    keypair: {
+        publicKey: 'REPLACE_WITH_PUBLIC_KEY',
+        secretKey: 'REPLACE_WITH_PRIVATE_KEY'
+    },
+    networkId: 'ae_uat', //replace with ae_mainnet for mainnet
+    nativeMode: true
+});
+```
+
+### 2. Reading the ExampleContract
+
+To read the contract code, add after the previously added lines:
+
+```
+// read the ExampleContract.aes from same directory
+const contract = fs.readFileSync('./ExampleContract.aes', 'utf-8');
+```
+
+### 3. Compiling the contract code
+
+To compile the contract code, add after the previously added lines:
+
+```
+// use the client to compile the contract code, catching eventual errors
+const compiled = await client.contractCompile(contract).catch(logError);
+
+// logging the bytecode output to the console to see the code work
+console.log('contract bytecode:', compiled.bytecode);
+```
+
+### 4. Deploying the contract bytecode to the blockchain
+
+To deploy the contract bytecode, add after the previously added lines:
+
+```
+// use the client to deploy the contract bytecode to the blockchain, catching eventual errors
+// we have to pass the initState where we choose 10000 as the number of available count and 15 as minimum price per item
+const deployed = await client.contractDeploy(compiled.bytecode, 'sophia', {initState: '(10000, 15)'}).catch(logError);
+
+// logging the contract adddress to the console, this is later used to call function of the contract
+console.log('deployed contract address:', deployed.address);
+```
+
+
+### 5. Calling the contract functions
+
+To call the `reserve` function of the deployed contract. It can be called by anyone using the blockchain. Add after the previously added lines:
```

Comment:
"Let's reserve an item in our contract. This can be done via contract call to our deployed smart contract. *Note* that this function is callable by anyone."